### PR TITLE
Admin Merchants (User Stories 24 - 31)

### PR DESCRIPTION
**Pull request by cariperi**
This PR adds functionality for the Admin Merchants Invoice and Admin Merchants Show pages, including:

- Adding views and page headers
- Displaying merchants by their status (enabled / disabled) and adding buttons to toggle this status
- Displaying Top 5 Merchants by Total Revenue and statistics about these merchants
- Ability to create and update merchants, from the admin pages (links on admin pages, functionality in merchants controller actions)

**Testing:**

- Model Tests are passing at 100%
- Feature Tests are all passing for the new functionality, and all existing functionality EXCEPT the merchant user stories where test data is still being generated in Rails Helper (11 failing tests).  Team is refactoring and this will be addressed by future PRs.

**Helper Methods Heads Up:**

This PR includes two helper methods in the ApplicationHelper file, to format integer cents values as dollars with a $ sign, and to format dates per one specific wireframe with a m/d/y format. These methods are tested in the spec/helpers/application_helper_spec file.

**Migration Change Heads Up:**

This PR includes the addition of a new field "is_enabled" to the Merchant table, as a boolean, with a default value of "false" (disabled).  Recommend that you `drop, create, migrate` and re-load CSV data after merging this before running any tests or your server. 

@kohyoungheon and I split this up and each worked on half of the stories - individual commits are listed below and individual issues are labeled in our Projects board.

### CODE REVIEW

File: config/routes.rb
Comment (kohyoungheon) [inline config/routes.rb:1]: I think we should eventually double check these routes so we can be sure we don't have any unnecessary paths floating around.

File: spec/models/merchant_spec.rb [hunk 1]
```diff
@@ -12,6 +12,131 @@
...
+      @invoice_item_5 = create(:invoice_item, invoice_id: @invoice_5.id, item_id: @item_5.id, quantity: 5, unit_price: 5)
+      @invoice_item_6 = create(:invoice_item, invoice_id: @invoice_5.id, item_id: @item_5.id, quantity: 2, unit_price: 2)#Testing merchant with multiple invoices
+      @invoice_item_7 = create(:invoice_item, invoice_id: @invoice_5.id, item_id: @item_5.id, quantity: 1, unit_price: 1)#Testing merchant with multiple invoices
+      @invoice_item_8 = create(:invoice_item, invoice_id: @invoice_6.id, item_id: @item_6.id, quantity: 6, unit_price: 6)
+      @transaction_1 = create(:transaction, invoice_id: @invoice_1.id, result: "success")
+      @transaction_2 = create(:transaction, invoice_id: @invoice_2.id, result: "success")
+      @transaction_3 = create(:transaction, invoice_id: @invoice_3.id, result: "success")
+      @transaction_4 = create(:transaction, invoice_id: @invoice_4.id, result: "success")
+      @transaction_5 = create(:transaction, invoice_id: @invoice_5.id, result: "success")
+      @transaction_6 = create(:transaction, invoice_id: @invoice_6.id, result: false)
```
Comment (kohyoungheon) [inline spec/models/merchant_spec.rb:53]: I forgot to add a comment. This line is testing to make sure failed transactions dont count

File: spec/features/admin/merchants/index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+    it "displays the total revenue generated next to each merchant name" do
+      visit admin_merchants_path
+
+      within("#top_5") do
+        expect(page).to have_content("Total Revenue: $60")
+        expect(page).to have_content("Total Revenue: $50")
+        expect(page).to have_content("Total Revenue: $40")
+        expect(page).to have_content("Total Revenue: $30")
+        expect(page).to have_content("Total Revenue: $20")
+      end
```
Comment (kohyoungheon) [inline spec/features/admin/merchants/index_spec.rb:257]: Thanks for fixing the formatting for my tests :)

**Review by kohyoungheon — APPROVED** [review]
Everything is looking good!

File: spec/models/merchant_spec.rb (hunk 1 above)
Comment (kohyoungheon) [reply spec/models/merchant_spec.rb:53]: line 53

File: config/routes.rb
Comment (cariperi) [reply config/routes.rb:1]: Agreed! Will add this to my running list of things to do before we submit.

**Pull request merged**